src/internet_identity/src/stats/event_stats.rs

```diff
@@ -389,6 +389,10 @@ fn update_aggregation<M: Memory, F>(
     let mut pruned_weight_by_key = pruned_events
         .iter()
         .filter_map(&aggregation_filter_map)
+        // zero weighted events are ignored when adjusting aggregations during pruning because they
+        // don't contribute any change to the aggregations hence pruning them has no effect on
+        // aggregations
+        .filter(|weighted_aggregation| weighted_aggregation.weight > 0)
```

Comment:
If you remove this, does the test still pass?

Reply:
Yes.